v2/anomalydetection/backend-setup.mdx

```diff
@@ -0,0 +1,683 @@
+        // ... your code here ...
+    }
+    
+    if(responseData.requestIdInfo) {
+        // handle stuff like bot detection, vpn detection, tor detection, etc
+        // ... your code here ...
+    }
+
+    return responseData;
+}
+
+
+
+SuperTokens.init({
+    // @ts-ignore
+    framework: "...",
+    // @ts-ignore
+    appInfo: { /*...*/ },
+    recipeList: [
+        Passwordless.init({
+            // ... other customisations ...
+            // highlight-start
+            override: {
+                apis: (originalImplementation: APIInterface): APIInterface => {
+                    return {
+                        ...originalImplementation,
+                        createCodePOST: async function (input: any) {
+                            // We need to have a request ID when creating a code in order to detect possible bots, suspicious IP addresses, etc.
+                            const requestId = (await input.options.req.getJSONBody()).requestId;
+                            if(!requestId) {
+                                // make sure to throw an error or correctly handle this error case
+                                throw new Error("requestId is required");
+                            }
+
+                            // ... your code here ...
+                            let session = await Session.getSession(input.options.req, input.options.res, {
+                                overrideGlobalClaimValidators: () => [],
+                            });
+                            let phoneNumber: string = session.getAccessTokenPayload().phoneNumber;
+
+                            // we check the anomaly detection service before calling the original implementation of signUp
+                            let securityCheckResponse = await handleSecurityChecks({ ...input, requestId, phoneNumber });
+                            // ... security check code here ...
+
+                            return originalImplementation.createCodePOST(input);
+                        },
+                        resendCodePOST: async function (input: any) {
+                            // We need to have a request ID when resending the code in order to detect possible bots, suspicious IP addresses, etc.
+                            const requestId = (await input.options.req.getJSONBody()).requestId;
+                            if(!requestId) {
+                                // make sure to throw an error or correctly handle this error case
+                                throw new Error("requestId is required");
+                            }
+
+                            // ... your code here ...
+                            let session = await Session.getSession(input.options.req, input.options.res, {
+                                overrideGlobalClaimValidators: () => [],
+                            });
+                            let phoneNumber: string = session.getAccessTokenPayload().phoneNumber;
```

Comment:
you can get the phone number using the input.preAuthSessionId, and then calling the Passwordless.listCodesByPreAuthSessionId function with it. The return will be an object that contains a phone number of an email